TF_Settings_Web/src/Pages/Visuals/VisualsScreen.tsx

```diff
@@ -117,22 +124,20 @@ const VisualsScreen: React.FC = () => {
                     </div>
                     <LabelledToggleSwitch
                         name="Enable Cursor"
-                        value={state.cursorEnabled}
-                        onChange={(value) => dispatch({ content: { cursorEnabled: value }, writeOutConfig: true })}
+                        value={config.current.cursorEnabled}
+                        onChange={(value) => updateConfig({ cursorEnabled: value })}
                     />
-                    {state.cursorEnabled && (
+                    {config.current.cursorEnabled && (
                         <>
                             <div className={classes('cursor-style')}>
                                 <RadioGroup
                                     name="StylePresets"
                                     selected={styleOptions.indexOf(currentPreset) ?? 0}
                                     options={styleOptions}
                                     onChange={(preset) => {
-                                        dispatch({
-                                            content: { activeCursorPreset: presets.indexOf(preset as CursorPreset) },
-                                            writeOutConfig: true,
-                                        });
+                                        updateConfig({ activeCursorPreset: presets.indexOf(preset as CursorPreset) });
                                         updateCursorPreview(cursorStyles[preset as CursorPreset]);
+                                        writeVisualsConfigIfNew();
```

Comment:
I added this as otherwise the config is updated `onChange` which happens after the `pointerup` event fires, so you end up trying to write the non-updated config.

Reply:
👍 Just tested and see what you mean